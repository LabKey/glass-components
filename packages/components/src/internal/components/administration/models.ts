interface Member {
    id: number;
    name: string;
    type: string;
}

interface Group {
    groupName: string;
    members: Member[];
}

export interface GroupMembership {
    [key: string]: Group;
}
