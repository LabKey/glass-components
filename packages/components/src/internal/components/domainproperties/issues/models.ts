/*
 * Copyright (c) 2020 LabKey Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { immerable } from 'immer';

import { DomainDesign } from '../models';

export interface IssuesListDefOptionsConfig {
    entityId?: string;
    issueDefName: string;
    singularItemName: string;
    pluralItemName: string;
    commentSortDirection: string;
    assignedToGroup: number;
    assignedToUser: number;
}

interface IssuesListDefModelConfig extends IssuesListDefOptionsConfig {
    exception: string;
    domain: DomainDesign;
    domainId: number;
    domainKindName: string;
}

export class IssuesListDefModel implements IssuesListDefModelConfig {
    [immerable] = true;

    readonly exception: string;
    readonly domain: DomainDesign;
    readonly entityId?: string;
    readonly domainId: number;
    readonly issueDefName: string;
    readonly singularItemName: string = 'Issue';
    readonly pluralItemName: string = 'Issues';
    readonly commentSortDirection: string = 'ASC';
    readonly assignedToGroup: number;
    readonly assignedToUser: number;
    readonly domainKindName: string;

    constructor(values?: Partial<IssuesListDefModelConfig>) {
        Object.assign(this, values);
    }

    static create(raw: any, defaultSettings = null): IssuesListDefModel {
        if (defaultSettings) {
            const domain = DomainDesign.create(undefined);
            return new IssuesListDefModel({ ...defaultSettings, domain });
        } else {
            const domain = DomainDesign.create(raw.domainDesign);
            return new IssuesListDefModel({ ...raw.options, domain });
        }
    }

    isNew(): boolean {
        return !this.entityId;
    }

    isValid(): boolean {
        return this.hasValidProperties() && !this.domain.hasInvalidFields();
    }

    hasValidProperties(): boolean {
        return this.issueDefName !== undefined && this.issueDefName !== null && this.issueDefName.trim().length > 0;
    }

    getOptions(): IssuesListDefOptionsConfig {
        return {
            entityId: this.entityId,
            issueDefName: this.issueDefName,
            singularItemName: this.singularItemName,
            pluralItemName: this.pluralItemName,
            commentSortDirection: this.commentSortDirection,
            assignedToGroup: this.assignedToGroup,
            assignedToUser: this.assignedToUser,
        };
    }
}
