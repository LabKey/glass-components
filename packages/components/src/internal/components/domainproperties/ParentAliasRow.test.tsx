import React from 'react';
import { act } from '@testing-library/react';

import { userEvent } from '@testing-library/user-event';

import { SELECT_INPUT_CONTROL_SELECTOR } from '../forms/input/SelectInputTestUtils';

import { IParentAlias, IParentOption } from '../entities/models';

import { renderWithAppContext } from '../../test/reactTestLibraryHelpers';

import { ParentAliasRow } from './ParentAliasRow';

describe('ParentAliasRow', () => {
    test('Ignore empty values', async () => {
        const parentAlias: IParentAlias = {
            alias: '',
            id: '',
            parentValue: undefined,
            ignoreAliasError: true,
            ignoreSelectError: true,
        };

        let container;
        await act(async () => {
            container = renderWithAppContext(
                <ParentAliasRow
                    id={parentAlias.id}
                    onAliasChange={jest.fn()}
                    onRemove={jest.fn()}
                    parentAlias={parentAlias}
                    parentOptions={[]}
                />
            );
        });

        expect(document.getElementsByName('alias')[0].getAttribute('value')).toBe('');
        expect(document.querySelectorAll('.has-error')).toHaveLength(0);
        expect(document.getElementsByName('required')[0].getAttribute('checked')).toBeNull();
    });

    test('Blank values, Error CSS applied', async () => {
        const parentAlias: IParentAlias = {
            alias: '',
            id: 'testId',
            parentValue: undefined,
            ignoreAliasError: false,
            ignoreSelectError: false,
        };

        let container;
        await act(async () => {
            container = renderWithAppContext(
                <ParentAliasRow
                    id={parentAlias.id}
                    onAliasChange={jest.fn()}
                    onRemove={jest.fn()}
                    parentAlias={parentAlias}
                    parentOptions={[]}
                />
            );
        });

        const aliasInput = document.getElementsByName('alias')[0];
        expect(aliasInput.getAttribute('value')).toBe(parentAlias.alias);
        expect(document.querySelectorAll('.has-error input[name="alias"]')).toHaveLength(1);
        expect(document.querySelectorAll(`.has-error ${SELECT_INPUT_CONTROL_SELECTOR}`)).toHaveLength(1);
        expect(document.getElementsByName('required')[0].getAttribute('checked')).toBeNull();
    });

    test('With values', async () => {
        const parentAlias: IParentAlias = {
            alias: 'testAlias',
            id: 'testId',
            ignoreAliasError: false,
            ignoreSelectError: false,
            parentValue: { value: 'materialInputs/test', label: 'Test Label' },
        };
        const option: IParentOption = {
            label: 'test',
            query: 'sampleset',
            schema: 'exp',
            value: 'materialInputs/test',
        };

        let container;
        await act(async () => {
            container = renderWithAppContext(
                <ParentAliasRow
                    id={parentAlias.id}
                    onAliasChange={jest.fn()}
                    onRemove={jest.fn()}
                    parentAlias={parentAlias}
                    parentOptions={[option]}
                />
            );
        });

        expect(container).toMatchSnapshot();

        expect(document.getElementsByName('alias')[0].getAttribute('value')).toBe(parentAlias.alias);
        expect(document.getElementsByName('parentValue')[0].getAttribute('value')).toBe(parentAlias.parentValue.value);
        expect(document.getElementsByName('required')[0].getAttribute('checked')).toBeNull();
    });

    test('With values - required', async () => {
        const parentAlias: IParentAlias = {
            alias: 'testAlias',
            id: 'testId',
            ignoreAliasError: false,
            ignoreSelectError: false,
            parentValue: { value: 'materialInputs/test', label: 'Test Label' },
            required: true,
        };
        const option: IParentOption = {
            label: 'test',
            query: 'sampleset',
            schema: 'exp',
            value: 'materialInputs/test',
        };

        let container;
        await act(async () => {
            container = renderWithAppContext(
                <ParentAliasRow
                    id={parentAlias.id}
                    onAliasChange={jest.fn()}
                    onRemove={jest.fn()}
                    parentAlias={parentAlias}
                    parentOptions={[option]}
                />
            );
        });

        expect(document.getElementsByName('alias')[0].getAttribute('value')).toBe(parentAlias.alias);
        expect(document.getElementsByName('parentValue')[0].getAttribute('value')).toBe(parentAlias.parentValue.value);
        expect(document.getElementsByName('required')[0].getAttribute('checked')).toBe('');
    });

    test('With parent value not an option in select', async () => {
        const parentAlias: IParentAlias = {
            alias: 'testAlias',
            id: 'testId',
            ignoreAliasError: false,
            ignoreSelectError: false,
            parentValue: { value: 'materialInputs/test', label: 'Test Label' },
        };
        const option: IParentOption = {
            label: 'test',
            query: 'sampleset',
            schema: 'exp',
            value: 'materialInputs/notFound',
        };

        let container;
        await act(async () => {
            container = renderWithAppContext(
                <ParentAliasRow
                    id={parentAlias.id}
                    onAliasChange={jest.fn()}
                    onRemove={jest.fn()}
                    parentAlias={parentAlias}
                    parentOptions={[option]}
                />
            );
        });

        expect(container).toMatchSnapshot();

        expect(document.getElementsByName('alias')[0].getAttribute('value')).toBe(parentAlias.alias);
        expect(document.getElementsByName('parentValue')[0].getAttribute('value')).toBe(parentAlias.parentValue.value);
        expect(document.getElementsByName('required')[0].getAttribute('checked')).toBeNull();
    });

    // await userEvent.click(screen.getByText('card1'));
    test('Simulate delete behavior', async () => {
        const parentAlias: IParentAlias = {
            alias: 'testAlias',
            id: 'testId',
            ignoreAliasError: false,
            ignoreSelectError: false,
            parentValue: { value: 'materialInputs/option2', label: 'Test Label' },
        };
        const option1: IParentOption = {
            label: 'option1',
            query: 'sampleset',
            schema: 'exp',
            value: 'materialInputs/option1',
        };
        const option2: IParentOption = {
            label: 'option2',
            query: 'sampleset',
            schema: 'exp',
            value: 'materialInputs/option2',
        };

        const mockRemove = jest.fn();

        let container;
        await act(async () => {
            container = renderWithAppContext(
                <ParentAliasRow
                    id={parentAlias.id}
                    onAliasChange={jest.fn()}
                    onRemove={mockRemove}
                    parentAlias={parentAlias}
                    parentOptions={[option1, option2]}
                />
            );
        });

        const removeBtn = document.querySelector('span.fa-times-circle');
        expect(mockRemove).toHaveBeenCalledTimes(0);
        await userEvent.click(removeBtn);
        expect(mockRemove).toHaveBeenCalledTimes(1);

        // Remove should be called with the ID to be removed
        expect(mockRemove).toHaveBeenCalledWith(parentAlias.id);
    });

    test('Simulate changed alias and check required', async () => {
        const parentAlias: IParentAlias = {
            alias: 'testAlias',
            id: 'testId',
            parentValue: { value: 'materialInputs/option2', label: 'Test Label' },
            ignoreAliasError: false,
            ignoreSelectError: false,
        };
        const option1: IParentOption = {
            label: 'option1',
            query: 'sampleset',
            schema: 'exp',
            value: 'materialInputs/option1',
        };
        const option2: IParentOption = {
            label: 'option2',
            query: 'sampleset',
            schema: 'exp',
            value: 'materialInputs/option2',
        };

        const mockChange = jest.fn();

        let container;
        await act(async () => {
            container = renderWithAppContext(
                <ParentAliasRow
                    id={parentAlias.id}
                    onAliasChange={mockChange}
                    onRemove={jest.fn()}
                    parentAlias={parentAlias}
                    parentOptions={[option1, option2]}
                    updateDupeParentAliases={jest.fn()}
                />
            );
        });

        const aliasInput = document.getElementsByName('alias')[0];
        const requiredInput = document.getElementsByName('required')[0];

        expect(aliasInput.getAttribute('value')).toBe(parentAlias.alias);
        expect(requiredInput.getAttribute('checked')).toBeNull();
        expect(mockChange).toHaveBeenCalledTimes(0);

        await userEvent.type(aliasInput, 'c');
        expect(mockChange).toHaveBeenCalledTimes(1);
        expect(mockChange).toHaveBeenCalledWith('testId', 'alias', parentAlias.alias + 'c');
        mockChange.mockClear();

        await userEvent.click(requiredInput);
        expect(mockChange).toHaveBeenCalledTimes(1);
        expect(mockChange).toHaveBeenCalledWith('testId', 'required', true);
        mockChange.mockClear();
    });
});
