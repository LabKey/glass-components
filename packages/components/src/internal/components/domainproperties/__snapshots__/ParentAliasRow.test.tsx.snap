// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParentAliasRow With parent value not an option in select 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row"
      >
        <div
          class="col-xs-2"
        />
        <div
          class="col-xs-10"
        >
          <div
            class="domain-field-alias--row domain-field-row domain-row-border-default"
          >
            <div
              class="col-xs-4 domain-field-alias--input"
            >
              <div
                class="select-input-container "
              >
                <div
                  class="import-alias--parent-select form-group"
                >
                  <div
                    class="select-input css-b62m3t-container"
                    id="selectinput-4"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-6-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                      role="log"
                    />
                    <div
                      class="select-input__control css-13cymwt-control"
                    >
                      <div
                        class="select-input__value-container select-input__value-container--has-value css-hlgwow"
                      >
                        <div
                          class="select-input__single-value css-1dimb5e-singleValue"
                        >
                          materialInputs/test
                        </div>
                        <div
                          class="select-input__input-container css-19bb58m"
                          data-value=""
                        >
                          <input
                            aria-activedescendant=""
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-haspopup="true"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class="select-input__input"
                            id="react-select-6-input"
                            role="combobox"
                            spellcheck="false"
                            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="select-input__indicators css-1wy0on6"
                      >
                        <div
                          aria-hidden="true"
                          class="select-input__indicator select-input__clear-indicator css-1xc3v61-indicatorContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-tj5bde-Svg"
                            focusable="false"
                            height="20"
                            viewBox="0 0 20 20"
                            width="20"
                          >
                            <path
                              d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                            />
                          </svg>
                        </div>
                        <span
                          class="select-input__indicator-separator css-1u9des2-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="select-input__indicator select-input__dropdown-indicator css-1xc3v61-indicatorContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-tj5bde-Svg"
                            focusable="false"
                            height="20"
                            viewBox="0 0 20 20"
                            width="20"
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="parentValue"
                      type="hidden"
                      value="materialInputs/test"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="col-xs-4 domain-field-alias--input"
            >
              <input
                class="form-control"
                name="alias"
                placeholder="Enter a parent alias for import"
                type="text"
                value="testAlias"
              />
            </div>
            <div
              class="col-xs-4 domain-field-alias--input"
            >
              <input
                name="required"
                type="checkbox"
              />
              <div
                class="pull-right"
              >
                <span
                  class="field-icon"
                  id="testId-delete"
                  title="Remove Parent Alias"
                >
                  <span
                    class="fa fa-times-circle domain-field-delete-icon"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
    >
      <div
        class="col-xs-2"
      />
      <div
        class="col-xs-10"
      >
        <div
          class="domain-field-alias--row domain-field-row domain-row-border-default"
        >
          <div
            class="col-xs-4 domain-field-alias--input"
          >
            <div
              class="select-input-container "
            >
              <div
                class="import-alias--parent-select form-group"
              >
                <div
                  class="select-input css-b62m3t-container"
                  id="selectinput-4"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-6-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    class="select-input__control css-13cymwt-control"
                  >
                    <div
                      class="select-input__value-container select-input__value-container--has-value css-hlgwow"
                    >
                      <div
                        class="select-input__single-value css-1dimb5e-singleValue"
                      >
                        materialInputs/test
                      </div>
                      <div
                        class="select-input__input-container css-19bb58m"
                        data-value=""
                      >
                        <input
                          aria-activedescendant=""
                          aria-autocomplete="list"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class="select-input__input"
                          id="react-select-6-input"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="select-input__indicators css-1wy0on6"
                    >
                      <div
                        aria-hidden="true"
                        class="select-input__indicator select-input__clear-indicator css-1xc3v61-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                          />
                        </svg>
                      </div>
                      <span
                        class="select-input__indicator-separator css-1u9des2-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="select-input__indicator select-input__dropdown-indicator css-1xc3v61-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="parentValue"
                    type="hidden"
                    value="materialInputs/test"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-xs-4 domain-field-alias--input"
          >
            <input
              class="form-control"
              name="alias"
              placeholder="Enter a parent alias for import"
              type="text"
              value="testAlias"
            />
          </div>
          <div
            class="col-xs-4 domain-field-alias--input"
          >
            <input
              name="required"
              type="checkbox"
            />
            <div
              class="pull-right"
            >
              <span
                class="field-icon"
                id="testId-delete"
                title="Remove Parent Alias"
              >
                <span
                  class="fa fa-times-circle domain-field-delete-icon"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ParentAliasRow With values 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row"
      >
        <div
          class="col-xs-2"
        />
        <div
          class="col-xs-10"
        >
          <div
            class="domain-field-alias--row domain-field-row domain-row-border-default"
          >
            <div
              class="col-xs-4 domain-field-alias--input"
            >
              <div
                class="select-input-container "
              >
                <div
                  class="import-alias--parent-select form-group"
                >
                  <div
                    class="select-input css-b62m3t-container"
                    id="selectinput-2"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-4-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                      role="log"
                    />
                    <div
                      class="select-input__control css-13cymwt-control"
                    >
                      <div
                        class="select-input__value-container select-input__value-container--has-value css-hlgwow"
                      >
                        <div
                          class="select-input__single-value css-1dimb5e-singleValue"
                        >
                          test
                        </div>
                        <div
                          class="select-input__input-container css-19bb58m"
                          data-value=""
                        >
                          <input
                            aria-activedescendant=""
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-haspopup="true"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class="select-input__input"
                            id="react-select-4-input"
                            role="combobox"
                            spellcheck="false"
                            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="select-input__indicators css-1wy0on6"
                      >
                        <div
                          aria-hidden="true"
                          class="select-input__indicator select-input__clear-indicator css-1xc3v61-indicatorContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-tj5bde-Svg"
                            focusable="false"
                            height="20"
                            viewBox="0 0 20 20"
                            width="20"
                          >
                            <path
                              d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                            />
                          </svg>
                        </div>
                        <span
                          class="select-input__indicator-separator css-1u9des2-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="select-input__indicator select-input__dropdown-indicator css-1xc3v61-indicatorContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-tj5bde-Svg"
                            focusable="false"
                            height="20"
                            viewBox="0 0 20 20"
                            width="20"
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="parentValue"
                      type="hidden"
                      value="materialInputs/test"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="col-xs-4 domain-field-alias--input"
            >
              <input
                class="form-control"
                name="alias"
                placeholder="Enter a parent alias for import"
                type="text"
                value="testAlias"
              />
            </div>
            <div
              class="col-xs-4 domain-field-alias--input"
            >
              <input
                name="required"
                type="checkbox"
              />
              <div
                class="pull-right"
              >
                <span
                  class="field-icon"
                  id="testId-delete"
                  title="Remove Parent Alias"
                >
                  <span
                    class="fa fa-times-circle domain-field-delete-icon"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
    >
      <div
        class="col-xs-2"
      />
      <div
        class="col-xs-10"
      >
        <div
          class="domain-field-alias--row domain-field-row domain-row-border-default"
        >
          <div
            class="col-xs-4 domain-field-alias--input"
          >
            <div
              class="select-input-container "
            >
              <div
                class="import-alias--parent-select form-group"
              >
                <div
                  class="select-input css-b62m3t-container"
                  id="selectinput-2"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-4-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    class="select-input__control css-13cymwt-control"
                  >
                    <div
                      class="select-input__value-container select-input__value-container--has-value css-hlgwow"
                    >
                      <div
                        class="select-input__single-value css-1dimb5e-singleValue"
                      >
                        test
                      </div>
                      <div
                        class="select-input__input-container css-19bb58m"
                        data-value=""
                      >
                        <input
                          aria-activedescendant=""
                          aria-autocomplete="list"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class="select-input__input"
                          id="react-select-4-input"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="select-input__indicators css-1wy0on6"
                    >
                      <div
                        aria-hidden="true"
                        class="select-input__indicator select-input__clear-indicator css-1xc3v61-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                          />
                        </svg>
                      </div>
                      <span
                        class="select-input__indicator-separator css-1u9des2-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="select-input__indicator select-input__dropdown-indicator css-1xc3v61-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="parentValue"
                    type="hidden"
                    value="materialInputs/test"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-xs-4 domain-field-alias--input"
          >
            <input
              class="form-control"
              name="alias"
              placeholder="Enter a parent alias for import"
              type="text"
              value="testAlias"
            />
          </div>
          <div
            class="col-xs-4 domain-field-alias--input"
          >
            <input
              name="required"
              type="checkbox"
            />
            <div
              class="pull-right"
            >
              <span
                class="field-icon"
                id="testId-delete"
                title="Remove Parent Alias"
              >
                <span
                  class="fa fa-times-circle domain-field-delete-icon"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
